components/producers/blackduck/main.go

```diff
@@ -0,0 +1,204 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	v1 "github.com/ocurity/dracon/api/proto/v1"
+
+	"github.com/ocurity/dracon/components/producers"
+	packageurl "github.com/package-url/packageurl-go"
+)
+
+func main() {
+	if err := producers.ParseFlags(); err != nil {
+		log.Fatal(err)
+	}
+
+	inFile, err := producers.ReadInFile()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var results BlackduckOut
+	if err := producers.ParseJSON(inFile, &results); err != nil {
+		log.Fatal(err)
+	}
+
+	issues, err := parseIssues(&results)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if err := producers.WriteDraconOut(
+		"gosec",
+		issues,
+	); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func cweIdsToInt(ids []string) []int32 {
+	res := []int32{}
+	for _, id := range ids {
+		newID := strings.ReplaceAll(id, "CWE-", "")
+		newIntID, err := strconv.Atoi(newID)
+		if err != nil {
+			log.Println(err)
+			continue
+		}
+		res = append(res, int32(newIntID))
+	}
+	return res
+}
+
+func BDTargetToPurl(originName, originID string) string {
+	splitOrigin := strings.Split(originID, ":")
```

Comment:
Note that some ecosystems don't have a concept of namespaces (Python, occasionally NPM, Rust, ...), so you might only get 2 parts in return for this `Split`.

Reply:
you mean something like this?